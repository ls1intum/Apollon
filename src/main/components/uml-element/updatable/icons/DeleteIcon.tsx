import React, { FunctionComponent } from 'react';

export const DeleteIcon: FunctionComponent<React.SVGProps<SVGSVGElement>> = (props) => (
  <svg {...props} height={20} width={20}>
    <path d="M4.66666 14C4.29999 14 3.9861 13.8694 3.72499 13.6083C3.46388 13.3472 3.33332 13.0333 3.33332 12.6667V4H2.66666V2.66667H5.99999V2H9.99999V2.66667H13.3333V4H12.6667V12.6667C12.6667 13.0333 12.5361 13.3472 12.275 13.6083C12.0139 13.8694 11.7 14 11.3333 14H4.66666ZM11.3333 4H4.66666V12.6667H11.3333V4ZM5.99999 11.3333H7.33332V5.33333H5.99999V11.3333ZM8.66666 11.3333H9.99999V5.33333H8.66666V11.3333Z" />
  </svg>
);
