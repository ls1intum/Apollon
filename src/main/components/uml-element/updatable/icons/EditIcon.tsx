import React, { FunctionComponent } from 'react';

export const EditIcon: FunctionComponent<React.SVGProps<SVGSVGElement>> = (props) => (
  <svg {...props} height={20} width={20}>
    <path d="M2.39999 13.6H3.53999L11.36 5.77999L10.22 4.63999L2.39999 12.46V13.6ZM0.799988 15.2V11.8L11.36 1.25999C11.52 1.11332 11.6967 0.999988 11.89 0.919988C12.0833 0.839988 12.2867 0.799988 12.5 0.799988C12.7133 0.799988 12.92 0.839988 13.12 0.919988C13.32 0.999988 13.4933 1.11999 13.64 1.27999L14.74 2.39999C14.9 2.54665 15.0167 2.71999 15.09 2.91999C15.1633 3.11999 15.2 3.31999 15.2 3.51999C15.2 3.73332 15.1633 3.93665 15.09 4.12999C15.0167 4.32332 14.9 4.49999 14.74 4.65999L4.19999 15.2H0.799988ZM10.78 5.21999L10.22 4.63999L11.36 5.77999L10.78 5.21999Z" />
  </svg>
);
