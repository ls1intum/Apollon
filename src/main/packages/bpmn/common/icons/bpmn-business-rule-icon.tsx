import React, { FunctionComponent } from 'react';
import { ThemedPolyline, ThemedRect } from '../../../../components/theme/themedComponents';

export const BPMNBusinessRuleIcon: FunctionComponent<React.SVGProps<SVGSVGElement>> = (props) => (
  <svg {...props} height={20} width={20}>
    <ThemedRect height={16} width={16} x={2} y={2} strokeLinejoin="round" fillColor="transparent" />
    <ThemedRect height={4} width={16} x={2} y={2} strokeLinejoin="round" fillColor="currentColor" />
    <ThemedPolyline points={'2 10, 18 10'} strokeLinecap="round" strokeLinejoin="round" />
    <ThemedPolyline points={'2 14, 18 14'} strokeLinecap="round" strokeLinejoin="round" />
    <ThemedPolyline points={'6 2, 6 18'} strokeLinecap="round" strokeLinejoin="round" />
  </svg>
);
