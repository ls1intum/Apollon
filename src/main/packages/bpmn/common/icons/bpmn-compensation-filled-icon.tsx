import React, { FunctionComponent } from 'react';
import { ThemedPolyline } from '../../../../components/theme/themedComponents';

export const BPMNCompensationFilledIcon: FunctionComponent<React.SVGProps<SVGSVGElement>> = (props) => (
  <svg {...props} height={20} width={20}>
    <ThemedPolyline
      points={'3 10, 9 6, 9 14, 3 10'}
      strokeLinecap="round"
      strokeLinejoin="round"
      fillColor="currentColor"
      fillRule="evenodd"
    />
    <ThemedPolyline
      points={'10 10, 16 6, 16 14, 10 10'}
      strokeLinecap="round"
      strokeLinejoin="round"
      fillColor="currentColor"
      fillRule="evenodd"
    />
  </svg>
);
