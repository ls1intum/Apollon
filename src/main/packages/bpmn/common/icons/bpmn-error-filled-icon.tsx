import React, { FunctionComponent } from 'react';
import { ThemedPolyline } from '../../../../components/theme/themedComponents';

export const BPMNErrorFilledIcon: FunctionComponent<React.SVGProps<SVGSVGElement>> = (props) => (
  <svg {...props} height={20} width={20}>
    <ThemedPolyline
      points={'3 16, 6 4, 13 11, 17 4, 14 16, 7 10, 3 16'}
      strokeLinecap="round"
      strokeLinejoin="round"
      fillColor="currentColor"
    />
  </svg>
);
