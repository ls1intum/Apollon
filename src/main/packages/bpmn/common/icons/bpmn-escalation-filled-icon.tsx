import React, { FunctionComponent } from 'react';
import { ThemedPolyline } from '../../../../components/theme/themedComponents';

export const BPMNEscalationFilledIcon: FunctionComponent<React.SVGProps<SVGSVGElement>> = (props) => (
  <svg {...props} height={20} width={20}>
    <ThemedPolyline
      points={'10 3, 4 15, 10 12, 16 15, 10 3'}
      strokeLinecap="round"
      strokeLinejoin="round"
      fillColor="currentColor"
      fillRule="evenodd"
    />
  </svg>
);
