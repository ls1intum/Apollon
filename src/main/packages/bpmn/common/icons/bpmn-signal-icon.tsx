import React, { FunctionComponent } from 'react';
import { ThemedPolyline } from '../../../../components/theme/themedComponents';

export const BPMNSignalIcon: FunctionComponent<React.SVGProps<SVGSVGElement>> = (props) => (
  <svg {...props} height={20} width={20}>
    <ThemedPolyline
      points={'10 3, 3 15, 17 15, 10 3'}
      strokeLinecap="round"
      strokeLinejoin="round"
      fillColor="transparent"
    />
  </svg>
);
