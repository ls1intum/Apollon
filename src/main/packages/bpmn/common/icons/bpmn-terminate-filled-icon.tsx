import React, { FunctionComponent } from 'react';
import { ThemedCircle } from '../../../../components/theme/themedComponents';

export const BPMNTerminateFilledIcon: FunctionComponent<React.SVGProps<SVGSVGElement>> = (props) => (
  <svg {...props} height={20} width={20}>
    <ThemedCircle cx="50%" cy="50%" r={10} fillColor="currentColor" />
  </svg>
);
