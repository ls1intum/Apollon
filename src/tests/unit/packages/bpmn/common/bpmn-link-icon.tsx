import { wrappedRender } from '../../../test-utils/render';
import * as React from 'react';
import { BPMNLinkIcon } from '../../../../../main/packages/bpmn/common/icons/bpmn-link-icon';

it('render the bpmn-link-icon', () => {
  const { baseElement } = wrappedRender(
    <svg>
      <BPMNLinkIcon />
    </svg>,
  );
  expect(baseElement).toMatchSnapshot();
});
