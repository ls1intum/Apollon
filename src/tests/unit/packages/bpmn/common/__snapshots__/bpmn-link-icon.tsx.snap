// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the bpmn-link-icon 1`] = `
<body>
  <div>
    <svg>
      <svg
        height="20"
        width="20"
      >
        <polyline
          class="sc-aXZVg dCdbOk"
          fill="transparent"
          points="3 7, 13 7, 13 4, 18 10, 13 16, 13 13, 3 13, 3 7"
          stroke="black"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </svg>
  </div>
</body>
`;
